qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/commons/ValueResolver.java

```diff
@@ -101,12 +114,35 @@ public String getSignature() {
 		return namespace != null ? namespace + ":" + signature : signature;
 	}
 
+	/**
+	 * Returns the resolved type if return type has generic (ex : T,
+	 * java.util.List<T>) by using the given java type argument.
+	 * 
+	 * @param argType argument Java type.
+	 * 
+	 * @return the resolved type if return type has generic (ex : T,
+	 *         java.util.List<T>) by using the given java type argument.
+	 */
+	@Override
+	public String resolveJavaElementType(ResolvedJavaTypeInfo argType) {
+		// Example with following signature:
+		// "orEmpty(arg : java.lang.Iterable<T>) : java.util.List<T>"
+		JavaParameterInfo parameter = getParameterAt(0); // arg : java.lang.Iterable<T>
+		if (parameter == null) {
+			return null;
+		}
+		return resolveReturnType(argType, parameter.getJavaType());
+	}
+
 	@Override
 	public String toString() {
 		ToStringBuilder b = new ToStringBuilder(this);
-		b.add("namespace", this.namespace);
+		b.add("namespace", this.getNamespace());

```

Comment:
Same as above: constants for Java type info

Reply:
You mean set "namespace" as constants. Let's keep like this, the toString is just called when you debug the instance.